src/vsr/replica.zig

```diff
@@ -6947,10 +6948,7 @@ pub fn ReplicaType(
         /// Replication to standbys works similarly, jumping off the replica just before primary.
         /// TODO Use recent heartbeat data for next replica to leapfrog if faulty (optimization).
         fn replicate(self: *Self, message: *Message.Prepare) void {
-            assert(self.status == .normal);
             assert(message.header.command == .prepare);
-            assert(message.header.view == self.view);
-            assert(message.header.op == self.op);
```

Comment:
can we assert `header.op >= self.op`?

Reply:
And I guess `header.view >= self.view`

Reply:
Re `header.view >= self.view` I agree that dropping messages with `header.op < self.view` should not be forwarded, since prepares from an older primary can never receive enough prepare_oks to get committed (and also for normal, view_change they are repairs).

Reply:
Re `header.op >= self.op`, I believe this would only be valid if we also check if status is normal _before_ replicating a prepare. However, I wast thinking that may it be safe, and rather beneficial to allow forwarding messages with `header.op < self.op` in some cases...

Consider a case in a 3-node cluster R0, R1, & R2, wherein R1 was primary, it prepared a bunch of ops but couldn't commit the last few. Then, R2 becomes primary and R0 has received its start_view and transitioned to normal, but R1 hasn't. Now, R2 sends a prepare to R1 (who is in view_change still), but it doesn't forward this prepare because it is in view_change status. In this case, it _could_ have been better for R1 to forward it along the chain, in the hope that the subsequent replicas are write it to their WAL (in this case R2, which is in normal and can process it) and prepare_ok it? 

Or was there a specific where you think it would be unsafe to do so?

Reply:
Good point re op! I guess there's also a case where in the new view we _trucated_ the ops, so even if our op is larger, its actually truncated, so it still non-repair.